public/app/features/dashboard/components/Inspector/PanelInspectActions.tsx

```diff
@@ -0,0 +1,82 @@
+import { PanelModel } from '../../state';
+
+export interface PanelInspectActionProps {
+  panel: PanelModel;
+  data?: PanelData;
+}
+
+export interface PanelInspectAction {
+  title: string;
+  description: string;
+
+  component: ComponentType<PanelInspectActionProps>;
+}
+
+export interface PanelInspectActionSupplier {
+  getActions: (panel: PanelModel) => PanelInspectAction[] | null;
+}
+
+const dummySupplier: PanelInspectActionSupplier = {
+  getActions: (panel: PanelModel) => {
+    return [
+      {
+        title: 'Do something',
+        description: 'that thingy',
+        // eslint-disable-next-line react/display-name
+        component: ({ panel }) => {
+          return (
+            <div>
+              DO THING ONE. Panel: <pre>{JSON.stringify(panel.targets)}</pre>
+            </div>
+          );
+        },
+      },
+      {
+        title: 'Do something else',
+        description: 'another thing',
+        // eslint-disable-next-line react/display-name
+        component: ({ panel }) => {
+          return (
+            <div>
+              DO THING TWO. Panel: <pre>{JSON.stringify(panel.targets)}</pre>
+            </div>
+          );
+        },
+      },
+    ];
+  },
+};
+
+(window as any).grafanaPanelInspectActionSupplier = dummySupplier;
```

Comment:
could we, as part of types, do something like this?

```
declare global {
  interface Window {
    grafanaPanelInspectActionSupplier?: PanelInspectActionSupplier
  }
}
```

Reply:
don't think this needs a global window scope, we can add a registry singletons in grafana/data that plugins registry actions